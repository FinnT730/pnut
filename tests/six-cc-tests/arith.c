void putstring(char *s) {
  while (*s) {
    putchar(*s);
    s = s + 1;
  }
}

void putnumber(int n) {
  int i = 0;
  int digits[10];
  if (n == 0) {
    putchar('0');
    return;
  }
  if (n < 0) {
    putchar('-');
    n = -n;
  }
  while (n > 0) {
    digits[i] = n % 10;
    n = n / 10;
    i = i + 1;
  }
  i = i - 1;
  while (i >= 0) {
    putchar(digits[i] + '0');
    i = i - 1;
  }
}

int main() {
  int x;
  int y;
  int z;

  x = 42;
  y = 128;

  z = ++x;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = --x;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x++;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x--;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x += y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x -= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);  
  z = x *= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x /= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x %= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x &= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x |= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x ^= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x <<= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = x >>= y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = ~z;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = +z;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = 1 ? x : y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);
  z = 0 ? x : y;
  putstring("x: ");
  putnumber(x);
  putchar(',');
  putchar(' ');
  putstring("y: ");
  putnumber(y);
  putchar(',');
  putchar(' ');
  putstring("z: ");
  putnumber(z);
  putchar(10);

  return 0;
}
