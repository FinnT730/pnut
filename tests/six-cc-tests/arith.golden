x: 43, y: 128, z: 43
x: 42, y: 128, z: 42
x: 43, y: 128, z: 42
x: 42, y: 128, z: 43
x: 170, y: 128, z: 170
x: 42, y: 128, z: 42
x: 5376, y: 128, z: 5376
x: 42, y: 128, z: 42
x: 42, y: 128, z: 42
x: 0, y: 128, z: 0
x: 128, y: 128, z: 128
x: 0, y: 128, z: 0
x: 0, y: 128, z: 0
x: 0, y: 128, z: 0
x: 0, y: 128, z: -1
x: 0, y: 128, z: -1
x: 0, y: 128, z: 0
x: 0, y: 128, z: 128
